checks/taskcluster/task_e2e.py

```diff
@@ -0,0 +1,240 @@
+                    success = False
+                    details["error"] = f"Message '{output_message}' not found in {url}"
+
+        except aiohttp.ClientError as e:
+            success = False
+            details["error"] = f"Failed to retrieve artifacts ({e})"
+
+        #
+        # 5. Report if it took too much time to be executed.
+        resolved = utils.utcfromisoformat(last_run["resolved"])
+        details["task"]["completed"] = resolved.isoformat()
+        duration = (resolved - created_at).seconds
+        details["task"]["duration"] = duration
+        if duration > max_duration:
+            success = False
+            details["error"] = f"Execution took too long ({duration}s)"
+
+        if age_task.seconds > task_lifetime:
+            # The task was completed a while ago.
+            # Let's create a new one, so that it has a chance to have
+            # its artifacts inspected in a next check run.
+            task_id = await create_and_index_task(
+                index, queue, queue_id, task_index_path, output_message
+            )
+            logger.info(
+                f"Last task was {age_task.seconds} secs old, created new task {task_id}"
+            )
+
+    else:
+        if age_task.seconds > max_duration:
+            # The task has been pending/running for too long.
+            success = False
+            details["error"] = f"Execution timeout ({age_task.seconds}s)"
+        # Otherwise, it's currently running, check is successful.
+
+    return success, details
+
+
+async def create_and_index_task(index, queue, queue_id, index_path, message):
+    """Create a task in the specified queue, and index it immediately.
+
+    The specified `message` will be ouput in the logs.
+    """
+    name = "end-to-end-test"
+    gen = taskcluster.stableSlugId()
+    task_id = gen(name)
+
+    now = datetime.utcnow()
+    in_3_hours = now + timedelta(hours=3)
+    in_1_year = now + timedelta(days=365)
+
+    payload = {
+        "taskQueueId": queue_id,
+        "created": now.isoformat(),
+        "deadline": in_3_hours.isoformat(),
+        "expires": in_1_year.isoformat(),
+        "payload": {
+            "command": [["/bin/bash", "-c", f'echo "{message}"']],
+            "maxRunTime": 600,
+        },
+        "metadata": {
+            "name": name,
+            "owner": "bwong-directs@mozilla.com",
+            "source": "https://github.com/mozilla-services/poucave",
+            "description": "A task for an end-to-end test ",
+        },
+    }
+    status = await queue.createTask(task_id, payload)
+    task_id = status["status"]["taskId"]
+
+    # Keep track of this task in the index, to make sure we only
+    # run one task at a time.
+    await index.insertTask(
+        index_path,
+        {
+            "taskId": task_id,
+            "data": {},
+            "expires": in_1_year.isoformat(),
+            "rank": int(utils.utcnow().timestamp()),
+        },
+    )
```

Comment:
I see now that you are putting the task directly in the index here. I think it might be simpler to add a `route` to the task definition to do this, so that the task definition is a little more self-explanatory, and that we only get completed tasks added to the index (which might otherwise be confusing, since usually only completed (resolved) tasks are stored in the index, I believe).